src/pages/CalendarPage/index.js

```diff
@@ -14,46 +16,105 @@ class CalendarPage extends React.Component {
-        venue: 'Frazier Station',
-        startDate: '2020-05-26T21:04:26+00:00',
-        endDate: '2020-05-26T22:04:26+00:00',
-      },
-    ];
-    this.setState({ events: calendarEvents });
+
+    getAllEvents().then(events => {
+      const calendarEvents = [];
+      events.forEach(newEventParam => {
+        // make a copy of the event
+        const newEvent = Object.assign(newEventParam);
+
+        // convert timestamp object to ISO time string
+        newEvent.startDate = newEvent.startDate.toDate().toISOString();
+        newEvent.endDate = newEvent.endDate.toDate().toISOString();
+
+        newEvent.title = newEvent.name;
+        newEvent.venue = newEvent.location;
+        calendarEvents.push(newEvent);
+      });
+      this.setState({ events: calendarEvents });
+    });
+
+    // const calendarEvents = [
+    //   {
+    //     id: 'qO8nJ50tCO57hptbxNZa',
+    //     title: 'Enhanced multi-tasking model',
+    //     description:
+    //       'Pressure someone institution fund account part. Entire couple develop main.\nMeeting sea school me policy beautiful well. Agent ground so majority care born blood.',
+    //     venue: 'Lindsey Rapids',
+    //     startDate: '2020-05-16T01:30:49+00:00',
+    //     endDate: '2020-05-16T02:30:49+00:00',
+    //   },
+    //   {
+    //     id: 'mRaXeYuSCMmqMOngDl7B',
+    //     title: 'Customizable bottom-line help-desk',
+    //     description:
+    //       'Reason clear rest the lay. Customer fill change.\nCampaign member we notice include investment. Near they order particularly western life. Reflect bed offer dinner top Mr of her.',
+    //     venue: 'Frazier Station',
+    //     startDate: '2020-03-30T21:04:26+00:00',
+    //     endDate: '2020-03-30T22:04:26+00:00',
+    //   },
+    // ];
+
+    // this.setState({ events: calendarEvents });
+  }
+
+  toggleView() {
+    this.setState(prevState => ({
+      view: prevState.view === 'calendar' ? ' list' : 'calendar',
+    }));
+
   }
 
   render() {
-    const { selectedEvent, events } = this.state;
+    const { selectedEvent, events, view } = this.state;
     const { classes } = this.props;
+
     return (
       <Grid className={classes.root} container spacing={1}>
+        <Grid container justify='flex-end'>
+          <Button
+            onClick={() => {
+              this.toggleView();
+            }}
+          >
+            {view === 'calendar' ? 'list View' : 'calendar view'}
+          </Button>
+        </Grid>
+
         <Grid item xs>
           <Paper>
-            <Calendar
-              events={events}
-              handleEventClick={event =>
-                this.setState({ selectedEvent: event })
-              }
-            />
+            {view === 'calendar' ? (
+              <Calendar
+                events={events}
+                handleEventClick={event => {
```

Comment:
We seem to be passing in the same function to the handleEventClick of both the EventList and Calendar components - can we pull that out and put it in its own function?

Reply:
done.